SimplioOfficial/simplio-app
# Task/45 searchable assets

# Overview

Searching assets is done using `SearchDelegate<T>` widget that uses local cubit to communicate with `AssetScreen` widget.
A search query filters assets by their `name` and `ticker`. Rebuilds are reduced by calling `_loadAsset` method on cubit when navigating to another screens.

## Test cases

- Click on floating add button
- Toggle an asset and check it toggle animation is smooth
- Go back and check if a wallet was created
- Click on floating add button again
- Toggle another asset
- Click on search in app bar and check if newly toggled asset is toggled in the asset list as well
- Search any asset by name or ticker
- Toggle searched asset
- Go back and check if the asset is toggled in listing
- Go back to dashboard and check if all wallets were created
